File: Algorithms/CyclicSort/CyclicSort.java
import java.util.Arrays;

public class CyclicSort {
    static int[] cyclicSort(int[] arr) {
        int i = 0;

        while (i < arr.length) {
            int correctIdx = arr[i] - 1;

            if (arr[i] != arr[correctIdx]) {
                int temp = arr[i];
                arr[i] = arr[correctIdx];
                arr[correctIdx] = temp;
            } else {
                i++;
            }
        }

        return arr;
    }

    public static void main(String[] args) {
        int[] arr = new int[] { 3, 4, 2, 1, 5 };
        int[] sortedArr = cyclicSort(arr);

        System.out.println(Arrays.toString(sortedArr));
    }

}

File: Algorithms/CyclicSort/CyclicSort.py
def cyclic_sort(arr):
    i = 0

    while i < len(arr):
        correct_idx = arr[i] - 1

        if arr[i] != arr[correct_idx]:
            arr[i], arr[correct_idx] = arr[correct_idx], arr[i]
        else:
            i += 1

    return arr


if __name__ == "__main__":
    arr = [3, 2, 1, 4, 5]
    print(cyclic_sort(arr))
